### DISCUSSION TITLE
The bundle version must be higher than the previously uploaded version

### DISCUSSION THREAD

**Discussion by paneiOS** [fastlane/fastlane#20743] [Q&A]
hello.
I have a problem uploading to testflight, please help me.

Currently, my app version 1.0.7 has been released.
I uploaded version 1.1.1(1) to testflight through fastlane.

The result of fastlane is failure.
### However, it is actually uploaded to testflight.
How did it happen?...

----------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------------

### MyApp Setting (in Xcode)
version : 1.1.1
Build : 1

----------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------------

### My Fastfile

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    increment_build_number(xcodeproj: "ComeHear.xcodeproj")
    build_app(
	scheme: "ComeHear",
	export_method: "app-store",
        export_options: {
        provisioningProfiles: { 
		"com.pane.ComeHear" => "ComeHear_fastLane"
	}
	})
    pilot(api_key_path: "fastlane/key.json")
    upload_to_testflight
  end
end

---------------------------------------------------------------------------------------------------------------------------------- 
----------------------------------------------------------------------------------------------------------------------------------

### My Result

[17:21:09]: [altool] 2022-10-14 17:21:09.917 *** Error: The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘2’. (ID: 11f3776f-4e27-4828-bbd5-a1ed6b8ba641) (-19232)

[17:21:09]: [altool]  {

[17:21:09]: [altool]     NSLocalizedDescription = "The provided entity includes an attribute with a value that has already been used";

[17:21:09]: [altool]     NSLocalizedFailureReason = "The bundle version must be higher than the previously uploaded version: \U20182\U2019. (ID: 11f3776f-4e27-4828-bbd5-a1ed6b8ba641)";

[17:21:09]: [altool]     "original_server_error" =     {

[17:21:09]: [altool]         code = "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE";

[17:21:09]: [altool]         detail = "The bundle version must be higher than the previously uploaded version.";

[17:21:09]: [altool]         id = "11f3776f-4e27-4828-bbd5-a1ed6b8ba641";

[17:21:09]: [altool]         meta =         {

[17:21:09]: [altool]             previousBundleVersion = 2;

[17:21:09]: [altool]         };

[17:21:09]: [altool]         source =         {

[17:21:09]: [altool]             pointer = "/data/attributes/cfBundleVersion";

[17:21:09]: [altool]         };

[17:21:09]: [altool]         status = 409;

[17:21:09]: [altool]         title = "The provided entity includes an attribute with a value that has already been used";

[17:21:09]: [altool]     };

[17:21:09]: [altool]     previousBundleVersion = 2;

[17:21:09]: [altool] }

[17:21:09]: Application Loader output above ^
[17:21:09]: Error uploading '/var/folders/fl/05lzx9790qz_g90v1t2v1pmr0000gn/T/a3b1c74b-1225-4852-9bfb-a75ab943dd90.ipa'.
[17:21:09]: The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘2’. (ID: 11f3776f-4e27-4828-bbd5-a1ed6b8ba641) (-19232)
[17:21:09]: The call to the altool completed with a non-zero exit status: 1. This indicates a failure.
[17:21:09]: Could not download/upload from App Store Connect!
+--------------------+------------------------------------------------------------------------------------------------+
|                                                    Lane Context                                                     |
+--------------------+------------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM   | ios                                                                                            |
| PLATFORM_NAME      | ios                                                                                            |
| LANE_NAME          | ios beta                                                                                       |
| BUILD_NUMBER       | 2                                                                                              |
| IPA_OUTPUT_PATH    | /Users/pane/Desktop/ComeHear/ComeHear.ipa                                                      |
| XCODEBUILD_ARCHIVE | /Users/pane/Library/Developer/Xcode/Archives/2022-10-14/ComeHear 2022-10-14 17.15.47.xcarchive |
| DSYM_OUTPUT_PATH   | /Users/pane/Desktop/ComeHear/ComeHear.app.dSYM.zip                                             |
+--------------------+------------------------------------------------------------------------------------------------+
[17:21:09]: Error uploading ipa file:
 [Application Loader Error Output]: Error uploading '/var/folders/fl/05lzx9790qz_g90v1t2v1pmr0000gn/T/a3b1c74b-1225-4852-9bfb-a75ab943dd90.ipa'.
[Application Loader Error Output]: The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘2’. (ID: 11f3776f-4e27-4828-bbd5-a1ed6b8ba641) (-19232)
[Application Loader Error Output]: The call to the altool completed with a non-zero exit status: 1. This indicates a failure.

+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | default_platform       | 0           |
| 2    | increment_build_number | 0           |
| 3    | build_app              | 45          |
| 4    | pilot                  | 276         |
| 💥   | upload_to_testflight   | 7           |
+------+------------------------+-------------+

[17:21:09]: fastlane finished with errors

[!] Error uploading ipa file:
 [Application Loader Error Output]: Error uploading '/var/folders/fl/05lzx9790qz_g90v1t2v1pmr0000gn/T/a3b1c74b-1225-4852-9bfb-a75ab943dd90.ipa'.
[Application Loader Error Output]: The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘2’. (ID: 11f3776f-4e27-4828-bbd5-a1ed6b8ba641) (-19232)
[Application Loader Error Output]: The call to the altool completed with a non-zero exit status: 1. This indicates a failure.

----------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------------

**Comment by ethanfischer** [comment]
Anyone solve this?

**Comment by AmadorNavarro** [comment]
Same problem. Seems to be a pilot error in 2.211.0 version. 2.204.0 works correctly.

**Reply by dion-gionet** [reply]
Tried to use 2.204.0 on my end but still doesn't work

**Comment by dion-gionet** [comment]
Same issue while trying to upload a macos build on testflight on fastlane 2.210.1

**Reply by mathaeus** [reply]
Same here with `2.210.1`, which worked fine up until recently, so in my opinion it can't be related to the `fastlane` version.
I assume it's Apple...

**Reply by triplef** [reply]
I also think this is a new behavior / check on Apple’s end. I tried uploading with Xcode and the Transporter app, and both gave me the same error.

Very unfortunate as it really complicates things for us with builds from multiple branches. I’ll open a Feedback Assistant ticket with Apple to try to revert that change, would be great if everyone here could do the same.

**Reply by dion-gionet** [reply]
I actually have a developer support request open for this, currently pending on a response. Hopefully they'll provide us with a solution or an alternative

**Reply by hakunatomata2** [reply]
any updates on this topic?

**Reply by dion-gionet** [reply]
Nope, I've had a really hard time getting answers from Apple so far. No one seems to be able to point me to the right direction, still trying though.

**Reply by svzi** [reply]
@dion-gionet Thank you for taking care. Please keep us noted.

**Reply by triplef** [reply]
Please all make sure to report this to Apple on your own as well via https://feedbackassistant.apple.com (or the Feedback Assitant macOS app). As this is a behavioral change that’s not strictly a bug, I think Apple will be unlikely to change it back unless they get significant feedback asking them to do so. Feel free to reference our ticket FB11871657.

Also make sure to include your app’s Apple ID in your feedback (Apple just came back with this request for our own ticket).

**Reply by dion-gionet** [reply]
So far, I've spoken to multiple Apple support engineers and they have not been able to pinpoint the problem yet. They have advised me to submit a bug report on feedback assistant and that they will keep me posted.

**Reply by GevaZeichner** [reply]
Is this now fixed? I think the issue was resolved for us starting Feb 9.

**Reply by triplef** [reply]
I can also confirm this is now working again for us. Uploading builds with build numbers that are lower than a previously uploaded build works again. 🎉

**Reply by triplef** [reply]
This issue is once again happening when using Xcode 15, whereas it worked fine for us using Xcode 14.3.1.

We’ve opened Feedback ticket FB13168491, feel free to submit your own referencing ours.

**Comment by faizeee** [comment]
Hi, i had the same problem, i am using **ionic** i resolve  this by updating `CURRENT_PROJECT_VERSION` in `ios/App/App.xcodeproj/project.pbxproj` file there are 2 locations you have to update. hopefully this will help someone.

**Reply by ajcentiblick** [reply]
This change resolved the issue for me while using Codemagic, thanks!

**Reply by kobimeridor** [reply]
Thanks! This change resolved the issue for me while using Codemagic

**Reply by CatStudioApp** [reply]
Nice spam.

**Reply by haqimRegov** [reply]
What is the format for app versioning that we need to follow or apply in `CURRENT_PROJECT_VERSION` ?

**Reply by faizeee** [reply]
it should be a number. i.e 1, 2, 3 every time you have new build just increment the number.

**Reply by Oleg-E-Bakharev** [reply]
It is very bad and unsafe idea to change and push to repo app project on build agent

**Comment by stephen-oconnor-cp** [comment]
Has anyone got an update on this?  I'm experiencing this issue, and I can't imagine such a glaring problem with the pipeline has no fix.

(Using fastlane 2.212.2)

**Comment by stephen-oconnor-cp** [comment]
The issue is that there are multiple ways to apply versioning to one's project.  In our case, with Apple Generic Versioning, you can't just update CURRENT_PROJECT_VERSION.  There are other spots in the project that 'update' the version.  This is really about making a fix for some changes Apple has made.

In our codebase, the "correct" way for us to bump the project version is to close the Xcode project, then run.

`xcrun agvtool next-version -all`

from the terminal.  (I presume agvtool is AppleGenericVersioningTOOL)

I think if one were to work on a fix, they should look into how Apple is doing versioning now as there are multiple ways they do this.  In our case we use Apple Generic Versioning and then as you see, the agvtool.

**Comment by mikemike396** [comment]
Failing again today using `2.213.0`. Anyone else seeing this?

**Reply by kurt-west** [reply]
I just started to have builds fail with this same error.

**Reply by kurt-west** [reply]
Not the idea case, but I used `xcrun agvtool next-version -all` to bump the major version number and the next build/push worked.

**Reply by chanda-is** [reply]
seeing this using version 2.214.0 - if I push the same ipa manually it works without bumping the version so it would seem to be something with fastlane/transporter

**Reply by tommyDoroGit** [reply]
Yep I'm getting the same error using version 2.214.0. I've bumped the version and build to numbers which clearly haven't been uploaded previously and still getting the error.

**Reply by stephen-oconnor-cp** [reply]
Yes, I believe the issue is with the way bumping version number is working.  I've had to remove this part from my pipeline and go in manually in a github action and run `xcrun agvtool next-version -all` to make it work.

**Comment by triplef** [comment]
This issue is once again happening when using Xcode 15, whereas it worked fine for us using Xcode 14.3.1.

We’ve opened Feedback ticket FB13168491, feel free to submit your own referencing ours.

**Reply by triplef** [reply]
It was once again fixed in the backend a month or so ago, so currently it’s working fine for us to upload builds with lower build numbers using Xcode 15.

**Comment by 0bmay** [comment]
for me, with Xcode 15.0.1 adding the direct path to the ipa file as a parameter to pilot worked for me.  Otherwise the file it was using was an older version in a random temp directory.

**Comment by ghost** [comment]
This just happened to me and none of the above solutions seems to be doing the trick. 
xCode Version 15.3 (15E204a)
fastlane 2.220.0

The specific error 

The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘6’. (ID: 824a29c2-cb90-4d3c-8894-78afb4f9a049) (-19232)

but my version is set to 9  and my build set to 1 (I can't find 6 anywhere) 
<img width="454" alt="image" src="https://github.com/fastlane/fastlane/assets/65786492/7c6d2c3a-34c1-4c0f-837d-1d6c67f5301a">


any suggestions appreciated.

**Reply by HETALvp** [reply]
I had same issue, for me `The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘7. `

I solved it by increasing build to 8 and upgrade version by +1 than  `Bundle Short Version String` on the latest app uploaded to Testflight

**Comment by Gloriaaaa23** [comment]
Thank you so much for your assistance!  [Office ally](https://officeally.fun/) Your help made a significant difference, and I truly appreciate it

**Comment by abdushkur** [comment]
version was indicated in two files, info.plist and project.pbxproj , my issue was on my fastlane increment_build_number skipped the plist file, it was set using Flutter_Build_Number variable , if we're going to use fastlane, don't skip plist file

**Comment by jaelynmasap** [comment]
I see this error too (which is not happening every time, but only to a large-size IPA upload). `The provided entity includes an attribute with a value that has already been used The bundle version must be higher than the previously uploaded version: ‘x.xx.x’. `
Fastlane version 2.220.0

**Comment by Kazutaka333** [comment]
I was using it for flutter project and I only run `flutter build ios --release` while getting the error. This was because it was trying to upload previously archived project.
So I just needed to rearchive the project with `build_app` lane with `export_method: "app-store"` before ruuning `upload_to_testflight`

**Comment by Nore1991** [comment]
Thank you for discussing this.
[Best Time to Visit Iceland](https://besttimetovisiticeland.com/)

**Comment by Ayeni-yeniyan** [comment]
I am using fastlane version: 2.232.2
fastlane for android works. This is a flutter project with build number 192 and other upload platforms pick the proper build number 
I am getting this error 
```bash

+--------------------+-------------------------------------------------------------------------------+
[00:06:10]: Called from Fastfile at line 166
[00:06:10]: ```
[00:06:10]:     164:          )
[00:06:10]:     165:
[00:06:10]:  => 166:          upload_to_testflight(
[00:06:10]:     167:            api_key: api_key,
[00:06:10]:     168:            skip_waiting_for_build_processing: true,
[00:06:10]: ```
[00:06:10]: Error uploading ipa file: 
 [Application Loader Error Output]: [ContentDelivery.Uploader.B26C39640] The provided entity includes an attribute with a value that has already been used (-19232) The bundle version must be higher than the previously uploaded version: ‘130’. (ID: my-id-placeholder)
[Application Loader Error Output]: [altool.B26C39640] Failed to upload package. 
```
